src/convertToObject.js

```diff
@@ -7,6 +7,18 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const elements = sourceString.split(';');
+  const result = {};
+
+  elements.forEach(function (element) {
+    const [name, value] = element.split(':');
+
+    if (name && value) {
+      result[name.trim()] = value.trim();
+    }
+  });
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. In this case, you are using forEach correctly to iterate over the elements array, but make sure not to modify the 'elements' array itself within the forEach callback.